readthedocs/core/static-src/core/js/doc-embed/search.js

```diff
@@ -7,6 +7,33 @@ var xss = require('xss/lib/index');
 var MAX_RESULT_PER_SECTION = 3;
 var MAX_SUBSTRING_LIMIT = 100;
 
+/**
+ * Use try...catch block to append html to contents
+ *
+ * @param {Object} contents html element on which additional html is be appended
+ * @param {String} template underscore.js template string
+ * @param {Object} data template vars and their values
+ */
+function append_html_to_contents(contents, template, data) {
+    // underscore.js throws variable not defined error
+    // because of change of syntax in new versions.
+    // See: https://stackoverflow.com/a/25881231/8601393
+    try {
+        // this is the old syntax
```

Comment:
Let's be a little more specific in the comment here. Sphinx bundles Underscore.js v1.3.1 (and has for many years) and it looks like this syntax changed in Underscore v1.7.0 (not exactly how semver is supposed to work...). So Sphinx should always use the old syntax and almost never the new syntax. How was this throwing a syntax error exactly? Were people somehow bundling newer versions of Underscore?

Reply:
Ok, I figured it out. Sphinx has bundled Underscore.js v1.3.1 as I mentioned. However, doc builds before ~2019 used centrally located versions of jQuery and Underscore.js (due to https://github.com/readthedocs/readthedocs-sphinx-ext/pull/52) as an optimization. This was removed precisely because it caused version mismatch issues like this!